### PR TITLE
feat: upgrade interchain stack to ibc5

### LINKED ISSUE #8719: MsgReturnGrants returns the wrong Type
## Describe the bug
In `agoric-labs/cosmos-sdk/x/auth/vesting/types/msgs.go`, `MsgReturnGrants.Type()` returns `TypeMsgClawback` instead of `TypeMsgReturnGrants`.

Impact is bad simulation of the command leading to a bad gas estimate.

## To Reproduce
See code.

## Expected behavior
See Description.

## Platform Environment
All.

### LINKED ISSUE #8783: agd reported incompatible with Ledger cosmos app
## Describe the bug
[Stakecito reported](https://discord.com/channels/585576150827532298/1004795902868852817/1198616247286366219) being unable to use the latest `agd` (currently at `agoric-upgrade-13`) with a Ledger, and provided a link to a [PR on another cosmos project](https://github.com/bitsongofficial/go-bitsong/pull/227) which also had the issue and solved it by bumping ledger-go related packages.

The issue as reported:
> the latest ag0/agd* binary is not compatible with Ledger anymore. The dependencies have to be upgraded. Could the team please look into that?

> txs do not work anymore with Ledger (e.g. tx authz). We just get an empty error:
> ```
> confirm transaction before signing and broadcasting [y/N]: y
> Error: 
> Usage: 
> ```

## To Reproduce
1. e.g. `agd tx authz` using Ledger

## Expected behavior
`agd` successfully signing tx using Ledger

## Platform Environment
 - `agoric-sdk` at `agoric-upgrade-13` release
 - Cosmos Ledger app at `v2.34.12`

## Additional context
- #8462 is already bumping `zondax/hid` to `v0.9.2`, and scheduled for inclusion in `agoric-upgrade-14`.
- We are currently using `v0.14.1` of `zondax/ledger-go`, and the [bump to `v.0.14.3`](https://github.com/Zondax/ledger-go/compare/v0.14.1...v0.14.3) seems to only be changing an error message and bumping the `zondax/hid` dep to `v0.9.2`.
- We are currently using `v0.12.2` of `cosmos/ledger-cosmos-go` and the [bump to `v0.12.3`](https://github.com/cosmos/ledger-cosmos-go/compare/v0.12.2...v0.12.4) seems to only be lowering the size of a message buffer, and bumping `zondax/hid` to `v0.9.2`
- The [changes to `zondax/hid`](https://github.com/Zondax/hid/compare/v0.9.1..v0.9.2) seem related only to MacOS detection issues

I suspect the actual fix is the message buffer size change in `cosmos/ledger-cosmos-go`. Regardless, updating these deps seems harmless.

### PR DESCRIPTION
**Pull request by JimLarson**
refs: #8224
closes: #8719
closes: #8783

## Description

Upgrade ibc-go to v5.3.2 (from v4.5.1) , with associated upgrades of cosmos-sdk (0.45.16 to 0.46.16) and cometbft (0.34.27 to 0.34.30).

### Security Considerations

Careful code review is required.

### Scaling Considerations

Performance testing is required.

### Documentation Considerations

N/A

### Testing Considerations

More testing notes will be added.

### Upgrade Considerations

No major upgrade-time work foreseen.

### CODE REVIEW

**Comment by JimLarson** [comment]
Made issue #8803 for some post-upgrade changes to remove use of deprecated APIs. We don't need to do them now, but we should do them before a migration where they go away.

**Comment by JimLarson** [comment]
Reviewers: please check the [cosmos-sdk](https://github.com/cosmos/cosmos-sdk/blob/v0.46.16/UPGRADING.md) and [ibc-go](https://github.com/cosmos/ibc-go/blob/v5.3.2/docs/migrations/v4-to-v5.md) migration guides as part of your review.

**Comment by JimLarson** [comment]
The remaining failing tests are expected failures. Protobuf / breakage because of protobuf compilation target changes, and golangci-lint because of use of deprecated APIs (see #8803 for later cleanup).

**Comment by JimLarson** [comment]
The cosmos-sdk module tag will be bumped to the final version once agoric-labs/cosmos-sdk#347 lands.

**Comment by JimLarson** [comment]
Adding @mhofman as an optional extra reviewer.

File: golang/cosmos/go.mod [hunk 1]
```diff
@@ -3,163 +3,194 @@ module github.com/Agoric/agoric-sdk/golang/cosmos
...
 
-// We need a fork of cosmos-sdk until all of the differences are merged.
-replace github.com/cosmos/cosmos-sdk => github.com/agoric-labs/cosmos-sdk v0.45.16-alpha.agoric.3
+	// https://pkg.go.dev/vuln/GO-2023-1578
+	github.com/hashicorp/go-getter => github.com/hashicorp/go-getter v1.7.0
+
+	// replace broken goleveldb.
+	github.com/syndtr/goleveldb => github.com/syndtr/goleveldb v1.0.1-0.20210819022825-2ae1ddf74ef7
+
+	// use cometft
```
Comment (michaelfig) [inline golang/cosmos/go.mod]: Just to prove I'm actually paying attention:

```suggestion
	// use cometbft
```

**Review by michaelfig — APPROVED** [review]
Everything looks kosher!  At least, I didn't find anything suspicious that a few minutes of search didn't clarify for me.

File: golang/cosmos/go.mod (hunk 1 above)
Comment (JimLarson) [reply golang/cosmos/go.mod]: Done. :-)

File: golang/cosmos/app/app.go [hunk 2]
```diff
@@ -560,6 +567,7 @@ func NewAgoricApp(
 		appCodec, keys[icahosttypes.StoreKey],
 		app.GetSubspace(icahosttypes.SubModuleName),
 		app.IBCKeeper.ChannelKeeper,
+		app.IBCKeeper.ChannelKeeper,
```
Comment (mhofman) [inline golang/cosmos/app/app.go:570]: Why the duplicated keeper here?

File: golang/cosmos/go.mod [hunk 3]
```diff
@@ -3,163 +3,194 @@ module github.com/Agoric/agoric-sdk/golang/cosmos
...
 	github.com/rs/cors v1.8.2 // indirect
-	github.com/rs/zerolog v1.27.0 // indirect
+	github.com/rs/zerolog v1.29.1 // indirect
 	github.com/sasha-s/go-deadlock v0.3.1 // indirect
 	github.com/spf13/afero v1.9.2 // indirect
 	github.com/spf13/jwalterweatherman v1.1.0 // indirect
 	github.com/spf13/pflag v1.0.5 // indirect
 	github.com/subosito/gotenv v1.4.1 // indirect
-	github.com/syndtr/goleveldb v1.0.1-0.20210819022825-2ae1ddf74ef7 // indirect
+	github.com/syndtr/goleveldb v1.0.1-0.20220721030215-126854af5e6d // indirect
```
Comment (mhofman) [inline golang/cosmos/go.mod:137]: Have we figured out what causes this one to be upgraded if cosmos-sdk didn't bump it?

File: golang/cosmos/go.mod [hunk 4]
```diff
@@ -3,163 +3,194 @@ module github.com/Agoric/agoric-sdk/golang/cosmos
...
+
+	// use cometbft
+	// Use our fork at least until post-v0.34.14 is released with
+	// https://github.com/tendermint/tendermint/issue/6899 resolved.
+	github.com/tendermint/tendermint => github.com/agoric-labs/cometbft v0.34.30-alpha.agoric.1
 
 // For testing against a local cosmos-sdk or tendermint
-// replace github.com/cosmos/cosmos-sdk => ../../../forks/cosmos-sdk
-// replace github.com/tendermint/tendermint => ../../../forks/tendermint
+// github.com/cosmos/cosmos-sdk => ../../../forks/cosmos-sdk
```
Comment (mhofman) [inline golang/cosmos/go.mod:194]: Why did the `replace` disappear here?

**Review by mhofman — COMMENTED** [review]
LGTM

File: golang/cosmos/go.mod (hunk 4 above)
Comment (JimLarson) [reply golang/cosmos/go.mod:194]: I changed it to a block construct, starting on line 166, to better match gaia.

File: golang/cosmos/app/app.go (hunk 2 above)
Comment (JimLarson) [reply golang/cosmos/app/app.go:570]: There's a new `ICS4Wrapper` parameter, for which the channel keeper is the correct default.

File: golang/cosmos/go.mod (hunk 3 above)
Comment (JimLarson) [reply golang/cosmos/go.mod:137]: I think it's some sub-dependency and/or the preference of selecting the latest compatible release. Cosmos-sdk has to override this version with a `replace` as well.

**Pull request merged**